### ISSUE TITLE
About evaluation of Dart-Math

### ISSUE THREAD

**Issue by XinXU-USTC** [hkust-nlp/dart-math#3]
Thank you for your excellent work and well-structured code repo.

I encountered some issues regarding the evaluation of Dart-Math. I ran the provided evaluation script to evaluate Dart-Math-Llama-3-8b-prof2diff on theoremqa and found that all items are false in the generated .jsonl file. What is the problem?

The resulted file is attached below.
https://hkustconnect-my.sharepoint.com/:u:/g/personal/xxuca_connect_ust_hk/ETFyWNbUrghFpa4H1_L6hpcB4AD8ze3mRoUrshPQcEGR3w?e=lzAEnX

**Comment by tongyx361** [comment]
Thanks for your issue! I figured out 2 bugs related to this error:

1. Commit https://github.com/hkust-nlp/dart-math/commit/a6647ea7e04c1d8c0e6b02a679d116de7c04f43d set `ignore_eos=True` because Llama-3-8B(-Base) tends to decode EoS immediately sometimes, but this messes up generation of normal (instruct) models. So I made `ignore_eos` a CLI option, which is `False` by default, in the latest commit https://github.com/hkust-nlp/dart-math/commit/82377fc66af5c1c76bf3d8e40175cd21772fa660.
2. I find that [the DART-Math-Llama3-8B (Prop2Diff) was wrongly uploaded as the checkpoint from epoch 3](https://huggingface.co/hkust-nlp/dart-math-llama3-8b-prop2diff-epoch3/tree/main), while we stated in the paper that we set # of epochs as 1 for Llama3 models for better performance found by hyperparameter search. I will upload [the correct checkpoint](https://huggingface.co/hkust-nlp/dart-math-llama3-8b-prop2diff/tree/main) asap (and close the issue then as a reminder).

Thanks again for helping us figure out the bugs!
